apps/src/codebridge/FilePreview/htmlParsingHelpers.ts

```diff
@@ -2,25 +2,40 @@ import {findFilePathByRelativePath} from '../utils';
 
 import {IframeMessageType} from './constants';
 
-// Replace links to non-html files (css and js) with their appropriate blob URLs.
-// We support <link> tags for CSS files and <script> tags for JavaScript files,
+// Replace links to non-html files (css and js) with their appropriate URLs (either blobs or external URLs).
+// We support <link> tags for CSS files, <script> tags for JavaScript files, and <img> tags for images,
 // and support both relative and absolute paths.
 export const updateLinksToNonHtmlFiles = (
   doc: Document,
-  filesToBlobs: Record<string, string>,
-  fullFileName: string
+  filesToUrls: Record<string, string>,
+  fullFileName: string,
+  parentOrigin: string
 ) => {
-  const links = doc.querySelectorAll('link[rel="stylesheet"], script[src]');
+  const imgLinks = doc.querySelectorAll('img[src]');
+  imgLinks.forEach(link => {
+    const src = link.getAttribute('src');
+
+    // Only update if the URL does not include a domain (eg, user project assets and starter assets)
+    if (src && !src.startsWith('http')) {
```

Comment:
Should we prevent image names from starting with `http` so we don't end up in a weird state here?

Reply:
Ooh good catch, can add a little something here...do we care if this limitation applies just to weblab2 or all of codebridge?

https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/codebridge/utils/validateFileName.ts

Reply:
Or just make this requirement starts with `http://` or `https://` since those wouldn't be valid file names?

Reply:
that's a better idea!